plugins/chat/assets/javascripts/discourse/components/chat/routes/channel-info-search.gjs

```diff
@@ -0,0 +1,68 @@
+import autoFocus from "discourse/modifiers/auto-focus";
+import Component from "@glimmer/component";
+import AsyncContent from "discourse/components/async-content";
+import { ajax } from "discourse/lib/ajax";
+import { bind } from "discourse/lib/decorators";
+import { i18n } from "discourse-i18n";
+import AdminFilterControls from "admin/components/admin-filter-controls";
+import AdminSectionLandingItem from "admin/components/admin-section-landing-item";
+import AdminSectionLandingWrapper from "admin/components/admin-section-landing-wrapper";
+import ChatMessage from "discourse/plugins/chat/discourse/models/chat-message";
+import ChatChannel from "discourse/plugins/chat/discourse/models/chat-channel";
+import ChatMessageComponent from "discourse/plugins/chat/discourse/components/chat-message";
+import { tracked } from "@glimmer/tracking";
+import FilterInput from "discourse/components/filter-input";
+import { action } from "@ember/object";
+import { hash } from "@ember/helper";
+
+export default class ChatRouteChannelInfoSearch extends Component {
+  @tracked term = "";
+
+  @bind
+  async searchMessages() {
+    const response = await ajax(
+      `/chat/api/channels/${this.args.channel.id}/messages/search`,
+      { data: { term: this.term, channel_id: this.args.channel.id } }
+    );
+
+    return (response.messages || []).map((messageObject) => {
+      const channel = ChatChannel.create(messageObject.channel);
```

Comment:
I think we can just do this once outside the map, since we are scoped to only 1 channel RN